pkg/clusterregistryconfig/flags.go

```diff
@@ -0,0 +1,369 @@
+		defaultBlockedRegistries = cluster.RegistryConfig().RegistrySources().BlockedRegistries()
+	}
+	if defaultInsecureRegistries == nil {
+		defaultInsecureRegistries = cluster.RegistryConfig().RegistrySources().InsecureRegistries()
+	}
+	if defaultAllowedRegistriesForImport == "" {
+		var list []string
+		for _, location := range cluster.RegistryConfig().AllowedRegistriesForImport() {
+			list = append(list, fmt.Sprintf("%s:%s", location.DomainName(), strconv.FormatBool(location.Insecure())))
+		}
+		defaultAllowedRegistriesForImport = strings.Join(list, ",")
+	}
+
+	enableRegistriesConfig := IsClusterRegistryConfigSetViaCLI(cmd)
+
+	if !enableRegistriesConfig && interactive.Enabled() && isHostedCP {
+		updateRegistriesConfigValue, err := interactive.GetBool(interactive.Input{
+			Question: "Enable registries config",
+			Default:  enableRegistriesConfig,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value: %s", err)
+		}
+		enableRegistriesConfig = updateRegistriesConfigValue
+	}
+
+	if enableRegistriesConfig && interactive.Enabled() {
+		allowedRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Allowed Registries",
+			Help:     cmd.Lookup("allowed-registries").Usage,
+			Default:  strings.Join(defaultAllowedRegistries, ","),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for allowed registries: %s", err)
+		}
+		result.allowedRegistries = helper.HandleEmptyStringOnSlice(strings.Split(allowedRegistriesInputs, ","))
+
+		blockedRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Blocked Registries",
+			Help:     cmd.Lookup("blocked-registries").Usage,
+			Default:  strings.Join(defaultBlockedRegistries, ","),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for blocked registries: %s", err)
+		}
+		result.blockedRegistries = helper.HandleEmptyStringOnSlice(strings.Split(blockedRegistriesInputs, ","))
+
+		insecureRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Insecure Registries",
+			Help:     cmd.Lookup("insecure-registries").Usage,
+			Default:  strings.Join(defaultInsecureRegistries, ","),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for insecure registries: %s", err)
+		}
+		result.insecureRegistries = helper.HandleEmptyStringOnSlice(strings.Split(insecureRegistriesInputs, ","))
+
+		result.allowedRegistriesForImport, err = interactive.GetString(interactive.Input{
+			Question: "Allowed Registries For Import",
+			Help:     cmd.Lookup("allowed-registries-for-import").Usage,
+			Default:  defaultAllowedRegistriesForImport,
+			Validators: []interactive.Validator{
+				ocm.ValidateAllowedRegistriesForImport,
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for allowed registries for import: %s", err)
+		}
+
+		result.additionalTrustedCa, err = interactive.GetString(interactive.Input{
```

Comment:
For certificates we usually supply the file and pull the information from it if I'm not mistaken
[Example](https://github.com/openshift/rosa/blob/965cea4de7c13531d682a6e650a111a76a6f611e/cmd/create/cluster/cmd.go#L2837) from additional trusted ca for proxy config

Reply:
hmm, I see this would not work as it is a json file with possibly many certificates, nvm